File: C++/Medium/WrongAnswer/stone-game-ix_8.cpp
class Solution {
public:
    bool stoneGameIX(vector<int>& stones) {
        
        int count[3] = {0, 0, 0};
        for (int x : stones) {
            count[x % 3]++;
        }
        return count[0] % 2 == 0 ? (count[1] && count[2]) : !(count[1] && count[2]);
    }
};

File: Java/Medium/Accepted/stone-game-ix_8.java
class Solution {
    public boolean stoneGameIX(int[] stones) {
        
        int[] bucket = new int[3];
        for (int stone : stones) {
            bucket[stone % 3]++;
        }
        if (bucket[0] % 2 == 0) {
            return bucket[1] > 0 && bucket[2] > 0;
        } else {
            return Math.abs(bucket[1] - bucket[2]) > 2;
        }
    }
}